src/software/thunderscope/binary_context_managers/game_controller.py

```diff
@@ -102,7 +107,23 @@ def refresh(self):
             )
             manual_command = self.command_override_buffer.get(return_cached=False)
 
-    def next_free_port(self, port: int = 40000, max_port: int = 65535) -> None:
+    def is_valid_port(self, port):
```

Comment:
nit: lets add typehinting
```suggestion
    def is_valid_port(self, port: int) -> bool:
```